src/sqlfluff/core/rules/std/L044.py

```diff
@@ -0,0 +1,140 @@
+"""Implementation of Rule L044."""
+from typing import Dict, List
+
+from sqlfluff.core.rules.analysis.select_crawler import SelectCrawler
+from sqlfluff.core.dialects.base import Dialect
+from sqlfluff.core.rules.base import BaseCrawler, LintResult
+
+
+class RuleFailure(Exception):
+    """Exception class for reporting lint failure inside deeply nested code."""
+
+    pass
+
+
+class Rule_L044(BaseCrawler):
+    """Query produces an unknown number of result columns.
+
+    | **Anti-pattern**
```

Comment:
Those explanations are gold, sqlfluff is really becoming a repository of knowledge for SQL

Reply:
Thanks! I recently reviewed the existing rules and found a few we may want to improve. #812